File: ts/384/350.ts
function intersect(nums1: number[], nums2: number[]): number[] {
  if (nums1.length > nums2.length) return intersect(nums2, nums1)
  const ht = {}, res = []
  for (const val of nums1) {
    if (ht[val]) {
      ht[val] += 1
    } else {
      ht[val] = 1
    }
  }
  for (const val of nums2) {
    if (ht[val] > 0) {
      ht[val] -= 1
      res.push(val)
    }
  }

  return res
};

File: py3/384/350.py
from typing import List

class Solution:
    def intersect(self, nums1: List[int], nums2: List[int]) -> List[int]:
        if len(nums1) > len(nums2):
            return self.intersect(nums2, nums1)
        ht = {}
        for v in nums1:
            if v in ht:
                ht[v] += 1
            else:
                ht[v] = 1
        ans = []
        for v in nums2:
            if v in ht:
                if ht[v] > 0:
                    ans.append(v)
                    ht[v] -= 1

        return ans